crates/test/src/rt/mod.rs

```diff
@@ -520,17 +573,39 @@ impl State {
         logs.push('\n');
     }
 
-    fn print_failure(&self, test: &Test, error: &JsValue) {
+    fn print_failure(&self, test: &Test, failure: &Failure) {
         let mut logs = String::new();
         let output = test.output.borrow();
+
+        match failure {
+            Failure::ShouldPanic => {
+                logs.push_str(&format!(
+                    "note: {} did not panic as expected\n\n",
+                    test.name
+                ));
+            }
+            Failure::ShouldPanicExpected => {
+                logs.push_str("note: panic did not contain expected string\n");
+                logs.push_str(&format!("      panic message: `\"{}\"`,\n", output.panic));
+                logs.push_str(&format!(
+                    " expected substring: `\"{}\"`\n\n",
+                    test.should_panic.unwrap().unwrap()
+                ));
+            }
+            _ => (),
+        }
+
         self.accumulate_console_output(&mut logs, "debug", &output.debug);
         self.accumulate_console_output(&mut logs, "log", &output.log);
         self.accumulate_console_output(&mut logs, "info", &output.info);
         self.accumulate_console_output(&mut logs, "warn", &output.warn);
         self.accumulate_console_output(&mut logs, "error", &output.error);
-        logs.push_str("JS exception that was thrown:\n");
-        let error_string = self.formatter.stringify_error(error);
-        logs.push_str(&tab(&error_string));
+
+        if let Failure::Error(error) = failure {
+            logs.push_str("JS exception that was thrown:\n");
+            let error_string = self.formatter.stringify_error(&error);
+            logs.push_str(&tab(&error_string));
+        }
```

Comment:
I could actually still print the error for the other two variants, but found it actually not that useful.